Core/src/game/functions/ints/count/sitesPlatformBelow/CountSitesPlatformBelow.java

```diff
@@ -0,0 +1,180 @@
+package game.functions.ints.count.sitesPlatformBelow;
+
+import java.util.BitSet;
+import java.util.List;
+
+import annotations.Hide;
+import annotations.Opt;
+import game.Game;
+import game.functions.ints.BaseIntFunction;
+import game.functions.ints.IntFunction;
+import game.functions.ints.iterator.To;
+import game.types.board.SiteType;
+import game.types.play.RoleType;
+import game.util.directions.AbsoluteDirection;
+import game.util.graph.Step;
+import main.Constants;
+import other.context.Context;
+import other.context.EvalContextData;
+import other.state.container.ContainerState;
+import other.topology.Topology;
+
+/**
+ * Returns the number of groups.
+ * 
+ * @author Eric.Piette & Cedric.Antoine
+ */
+@Hide
+public final class CountSitesPlatformBelow extends BaseIntFunction
+{
+	private static final long serialVersionUID = 1L;
+
+	//-------------------------------------------------------------------------
+
+	/** The graph element type. */
+	private SiteType type;
+
+	/** The owner of the pieces to make a line */
+	private final IntFunction whoFn;
+	
+	/** The site to test. */
+	private final IntFunction siteFn;
+	
+
+	//-------------------------------------------------------------------------
+
+	/**
+	 * @param who  		Player id the counted items belong to
+	 * @param site 		The site to check [(to)].
+	 */
+	public CountSitesPlatformBelow
+	(
+		@Opt 		final IntFunction site,
+					final RoleType who
+	)
+	{
+		siteFn = (site == null) ? To.instance() : site;
+		whoFn = RoleType.toIntFunction(who);
+	}
+
+	//-------------------------------------------------------------------------
+
+	@Override
+	public int eval(final Context context)
+	{
+		final int site = siteFn.eval(context);
+
+		if (site == Constants.OFF && site >= context.topology().vertices().size())
+			return -1;
+
+		final other.topology.Vertex v = context.topology().vertices().get(site);
+		final Topology topology = context.topology();
+
+		if (v.layer() == 0)
+			return 0;
+		
+		final ContainerState cs = context.containerState(context.containerId()[site]);
+		
+		final List<game.util.graph.Step> steps = topology.trajectories().steps(SiteType.Vertex, site, SiteType.Vertex,
+				AbsoluteDirection.Downward);
+		
+		int count = 0;
+		
+//		System.out.println("--------------");
+
+		for (final Step step : steps) {
+//			System.out.println("Field: " + step.to().id() + ", and id: " + cs.what(step.to().id(), SiteType.Vertex));
+//			System.out.println("RoleType: " + whoFn.eval(context));
+			if (cs.who(step.to().id(), SiteType.Vertex) == whoFn.eval(context)) {
+//				System.out.println();
+				count++;
+			}
+		}
+
+		return count;
+	}
+	
+	//-------------------------------------------------------------------------
+
+	@Override
+	public boolean isStatic()
+	{
+		return false;
+	}
+
+	@Override
+	public String toString()
+	{
+		return "Groups()";
+	}
+
+	@Override
+	public long gameFlags(final Game game)
+	{
+		long gameFlags = SiteType.gameFlags(type);
```

Comment:
this function, and most of the other functions below, should do everything they're doing with the two new members that are actually being used (and `type` should be removed if it's unused)